src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -226,6 +227,9 @@ def launch_fluent(
     The allocated machines and core counts are queried from the scheduler environment and
     passed to Fluent.
     """
+    if version and gpu:
+        if version != "3d" and gpu:
+            raise GPUSolverSupportError()
```

Comment:
@hpohekar, please add a test for this.

Reply:
Why two ifs like this? Why not just the second one?

Reply:
> @hpohekar, please add a test for this.

Done. Thank you.